x/twap/api_test.go

```diff
@@ -95,12 +120,34 @@ type getTwapInput struct {
 	endTime         time.Time
 }
 
-func makeSimpleTwapInput(startTime time.Time, endTime time.Time, isQuoteTokenA bool) getTwapInput {
+func makeSimpleTwapInput(startTime time.Time, endTime time.Time, isQuoteTokenA bool) []getTwapInput {
+	var twapInput []getTwapInput
 	quoteAssetDenom, baseAssetDenom := denom0, denom1
 	if isQuoteTokenA {
 		baseAssetDenom, quoteAssetDenom = quoteAssetDenom, baseAssetDenom
 	}
-	return getTwapInput{1, quoteAssetDenom, baseAssetDenom, startTime, endTime}
+	twapInput = append(twapInput, getTwapInput{1, quoteAssetDenom, baseAssetDenom, startTime, endTime})
+	return twapInput
+}
+
+func makeSimpleTapTwapInput(startTime time.Time, endTime time.Time, isQuoteTokenA bool) []getTwapInput {
+	var twapInput []getTwapInput
+	quoteAssetDenom, baseAssetDenom := denom0, denom1
+	if isQuoteTokenA {
+		baseAssetDenom, quoteAssetDenom = quoteAssetDenom, baseAssetDenom
+	}
+	twapInput = append(twapInput, getTwapInput{2, quoteAssetDenom, baseAssetDenom, startTime, endTime})
+	quoteAssetDenom, baseAssetDenom = denom0, denom2
+	if isQuoteTokenA {
+		baseAssetDenom, quoteAssetDenom = quoteAssetDenom, baseAssetDenom
+	}
+	twapInput = append(twapInput, getTwapInput{2, quoteAssetDenom, baseAssetDenom, startTime, endTime})
+	quoteAssetDenom, baseAssetDenom = denom1, denom2
```

Comment:
Is there a reason we're switching these two although either of the denom1 and denom2 are not tokenA? Do you think we should be either creating a secondary parameter for this function isQuoteTokenC or have this as a comment?

Reply:
Ah I see why this is confusing, good point. In my mind I was thinking TokenA as in the first lexicographical token of the pair, but it could also be seen literally as tokenA. Fixed this here, lmk if you like the implementation (its also more customizable which is desirable)

https://github.com/osmosis-labs/osmosis/pull/2604/commits/0d9d165ecdcdc1839e895cf62f9642da957b7177